### ISSUE TITLE
Side navigation drawer on small screens doesn't trap the focus

### ISSUE THREAD

**Issue by bartaz** [canonical/vanilla-framework#4560]
via @petermakowski comment via @petermakowski comment https://github.com/canonical/juju.is/pull/431#issuecomment-1231665391

> [...] the side navigation is missing a focus trap (if you keep pressing `Tab` you'll eventually start tabbing through elements on the page outside of it).

**Describe the bug**

The side navigation drawer "popup" on small screen doesn't trap the focus. Ideally it should move back up to "Toggle" side navigation button on top once you focus out of last link in the side nav (but only on small screens where side nav is collapsible).
